In [1]:
import pandas as pd

In machine learning you will most likely use databases from many sources to train your learning algorithms. Pandas allows us to load databases of different formats into DataFrames. One of the most popular data formats used to store databases is csv. CSV stands for *Comma Separated Values* and offers a simple format to store data. We can load CSV files into Pandas DataFrames using the `pd.read_csv()` function.

We will load the Google Stock Data into a DataFrame. The GOOG.csv is in the data folder of this notebook. It contains the stock prices of Google from 2004 to 2017, taken from Yahoo Finance.

#### Example 1. Load the data from a .csv file.

In [2]:
# We load Google stock data in a DataFrame
Google_stock = pd.read_csv('data/GOOG.csv')

# We print some information about Google_stock
print('Google_stock is of type:', type(Google_stock))
print('Google_stock has shape:', Google_stock.shape)

Google_stock is of type: <class 'pandas.core.frame.DataFrame'>
Google_stock has shape: (3313, 7)


We see that we have loaded the GOOG.csv file into a Pandas DataFrame and it consists of 3,313 rows and 7 columns. Now let's look at the stock data

#### Example 2. Look at the first few rows of the DataFrame

In [3]:
Google_stock

Date        Open        High         Low       Close   Adj Close  \
0     2004-08-19   49.676899   51.693783   47.669952   49.845802   49.845802   
1     2004-08-20   50.178635   54.187561   49.925285   53.805050   53.805050   
2     2004-08-23   55.017166   56.373344   54.172661   54.346527   54.346527   
3     2004-08-24   55.260582   55.439419   51.450363   52.096165   52.096165   
4     2004-08-25   52.140873   53.651051   51.604362   52.657513   52.657513   
...          ...         ...         ...         ...         ...         ...   
3308  2017-10-09  980.000000  985.424988  976.109985  977.000000  977.000000   
3309  2017-10-10  980.000000  981.570007  966.080017  972.599976  972.599976   
3310  2017-10-11  973.719971  990.710022  972.250000  989.250000  989.250000   
3311  2017-10-12  987.450012  994.119995  985.000000  987.830017  987.830017   
3312  2017-10-13  992.000000  997.210022  989.000000  989.679993  989.679993   

        Volume  
0     44994500  
1     23005800  
2     18393200  
3     15361800  
4      9257400  
...        ...  
3308    891400  
3309    968400  
3310   1693300  
3311   1262400  
3312   1157700  

[3313 rows x 7 columns]

We see that it is quite a large dataset and that Pandas has automatically assigned numerical row indices to the DataFrame. Pandas also used the labels that appear in the data in the CSV file to assign the column labels.

When dealing with large datasets like this one, it is often useful just to take a look at the first few rows of data instead of the whole dataset. We can take a look at the first 5 rows of data using the `.head()` method, as shown below

In [4]:
Google_stock.head()

Date       Open       High        Low      Close  Adj Close    Volume
0  2004-08-19  49.676899  51.693783  47.669952  49.845802  49.845802  44994500
1  2004-08-20  50.178635  54.187561  49.925285  53.805050  53.805050  23005800
2  2004-08-23  55.017166  56.373344  54.172661  54.346527  54.346527  18393200
3  2004-08-24  55.260582  55.439419  51.450363  52.096165  52.096165  15361800
4  2004-08-25  52.140873  53.651051  51.604362  52.657513  52.657513   9257400

We can also take a look at the last 5 rows of data by using the `.tail()` method:

#### Example 4. Look at the last 5 rows of the DataFrame

In [6]:
Google_stock.tail()

Date        Open        High         Low       Close   Adj Close  \
3308  2017-10-09  980.000000  985.424988  976.109985  977.000000  977.000000   
3309  2017-10-10  980.000000  981.570007  966.080017  972.599976  972.599976   
3310  2017-10-11  973.719971  990.710022  972.250000  989.250000  989.250000   
3311  2017-10-12  987.450012  994.119995  985.000000  987.830017  987.830017   
3312  2017-10-13  992.000000  997.210022  989.000000  989.679993  989.679993   

       Volume  
3308   891400  
3309   968400  
3310  1693300  
3311  1262400  
3312  1157700

We can also optionally use `.head(N)` or `.tail(N)` to display the first and last `N` rows of data, respectively.

Let's do a quick check to see whether we have any `NaN` values in our dataset. To do this, we will use the `.isnull()` method followed by the `.any()` method to check whether any of the columns contain `NaN` values.

#### Example 5. Check if any column contains a NaN. Returns a boolean for each column label.

In [10]:
Google_stock.isnull().any()

Date         False
Open         False
High         False
Low          False
Close        False
Adj Close    False
Volume       False
dtype: bool

There are no `NaN` values in our dataset.

When dealing with large datasets, it is often useful to get statistical information about the dataset. Pandas 